### PR TITLE
Logged in Customer placed order details endpoint

### PR DESCRIPTION
**Pull request by Zales0123**
Based on https://github.com/Sylius/ShopApiPlugin/pull/314

### CODE REVIEW

File: src/Factory/PlacedOrderViewFactory.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        $this->orderItemFactory = $orderItemFactory;
+        $this->addressViewFactory = $addressViewFactory;
+        $this->totalViewFactory = $totalViewFactory;
+        $this->shipmentViewFactory = $shipmentViewFactory;
+        $this->paymentViewFactory = $paymentViewFactory;
+        $this->adjustmentViewFactory = $adjustmentViewFactory;
+        $this->placedOrderViewClass = $placedOrderViewClass;
+    }
+
+    /** {@inheritdoc} */
```
Comment (lchrusciel) [inline src/Factory/PlacedOrderViewFactory.php]: Do we need it?

File: src/ViewRepository/PlacedOrderViewRepository.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        OrderRepositoryInterface $orderRepository,
+        CustomerRepositoryInterface $customerRepository,
+        PlacedOrderViewFactoryInterface $placedOrderViewFactory
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->customerRepository = $customerRepository;
+        $this->placedOrderViewFactory = $placedOrderViewFactory;
+    }
+
+    public function getCompletedByCustomerEmail(string $customerEmail): array
```
Comment (lchrusciel) [inline src/ViewRepository/PlacedOrderViewRepository.php]: `getCompletedByCustomerEmail` => `getAllCompletedByCustomerEmail`?

File: src/ViewRepository/PlacedOrderViewRepository.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        foreach ($this->orderRepository->findBy(['customer' => $customer]) as $order) {
+            if ($order->getCheckoutState() === OrderCheckoutStates::STATE_COMPLETED) {
+                $cartViews[] = $this->placedOrderViewFactory->create($order, $order->getLocaleCode());
+            }
+        }
+
+        return $cartViews;
+    }
+
+    public function getCompletedByCustomerEmailAndId(string $customerEmail, int $id): PlacedOrderView
```
Comment (lchrusciel) [inline src/ViewRepository/PlacedOrderViewRepository.php]: `getCompletedByCustomerEmailAndId` => `getOneCompletedByCustomerEmailAndId`

File: tests/Responses/Expected/order/order_not_found_response.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "code": 404,
+    "message": "There is no placed order with with id 13 for customer with email oliver@queen.com"
```
Comment (lchrusciel) [inline tests/Responses/Expected/order/order_not_found_response.json]: Only id 13?

File: src/Factory/PlacedOrderViewFactory.php (hunk 1 above)
Comment (Zales0123) [reply src/Factory/PlacedOrderViewFactory.php]: Nope 👍

File: spec/Factory/PlacedOrderViewFactorySpec.php [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+final class PlacedOrderViewFactorySpec extends ObjectBehavior
+{
+    public function let(
+        CartItemViewFactoryInterface $cartItemViewFactory,
+        AddressViewFactoryInterface $addressViewFactory,
+        TotalViewFactoryInterface $totalViewFactory,
+        ShipmentViewFactoryInterface $shipmentViewFactory,
+        PaymentViewFactoryInterface $paymentViewFactory,
+        AdjustmentViewFactoryInterface $adjustmentViewFactory
+    ) {
```
[the remark this answers is not in the document]
Comment (Zales0123) [reply spec/Factory/PlacedOrderViewFactorySpec.php]: I just wanted to be consistent with other specs. But I totally agree :)

File: tests/Responses/Expected/order/order_not_found_response.json (hunk 4 above)
Comment (Zales0123) [reply tests/Responses/Expected/order/order_not_found_response.json]: Nope, but this is an ID used in a test. Can we somehow make it dynamic?

**Comment by lchrusciel** [comment]
Could you rebase to the master? It will be easier to review

File: tests/Responses/Expected/order/order_not_found_response.json (hunk 4 above)
Comment (lchrusciel) [reply tests/Responses/Expected/order/order_not_found_response.json]: `@integer@`

**Comment by lchrusciel** [comment]
We need documentation as well? Ref: #334

**Comment by Zales0123** [comment]
We need for sure 👍 on it!

**Comment by Zales0123** [comment]
@lchrusciel done 📖

**Comment by lchrusciel** [comment]
Thanks Mati! 🐋

**Pull request merged**